### ISSUE TITLE
Batchnorm support for tracking buffer statistics when using gradient accumulation 

### ISSUE THREAD

**Issue by mathewhuen** [pytorch/pytorch#46516]

**Comment by mathewhuen** [comment]
It's my first time posting, so I apologize if I made any mistakes. Thank you in advance for any input!

**Comment by meder-yembo** [comment]
Maybe I'm oversimplifying, but it seems to me that some variation of [`SyncBatchNorm`](https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html) could apply here. It's effectively the same idea: processing subsets of the batch separately while maintaining the batch statistics of the whole batch. In this case, each iteration of a for loop stands in for a GPU. I haven't looked into the code to see how to modify SyncBatchNorm for this case, but I'd imagine the algorithm ought to be the same.

**Comment by meder411** [comment]
Maybe I'm oversimplifying, but it seems to me that some variation of SyncBatchNorm could apply here. It's effectively the same idea: processing subsets of the batch separately while maintaining the batch statistics of the whole batch. In this case, each iteration of a for loop stands in for a GPU. I haven't looked into the code to see how to modify SyncBatchNorm for this case, but I'd imagine the algorithm ought to be the same.

**Comment by trip036** [comment]
I'll be the first to admit I don't necessarily believe in being what's been
called an over achiever in the past, it was a great compliment in my
opinion.
